### PR TITLE
Reorganize docs to have friendlier URL paths 

### PR DESCRIPTION
**Pull request by jdbaldry**
Fix broken links that are currently relying on redirects and anchors that are out of date.

### CODE REVIEW

File: docs/sources/contribute/_index.md [hunk 1]
```diff
@@ -133,7 +135,7 @@ New users or members of other teams are very helpful for these tests.
 Prior to pushing your changes to GitHub, you can view a local build of the documentation so that you can review your work.
 For more information on using Git, refer to [Use Git](https://grafana.com/docs/writers-toolkit/write/tooling-and-workflows/#use-git).
 
-To view a local build, refer to [Test documentation changes](https://grafana.com/docs/writers-toolkit/review/run-a-local-webserver/#test-documentation-changes)
+To view a local build, refer to [Test documentation changes](https://grafana.com/docs/writers-toolkit/review/test-documentation-changes)
```
Comment (jdbaldry) [inline docs/sources/contribute/_index.md:138]:
```suggestion
To view a local build, refer to [Test documentation changes](https://grafana.com/docs/writers-toolkit/review/test-documentation-changes/)
```

File: docs/sources/write/reuse-content/reuse-shared-content/index.md [hunk 2]
```diff
@@ -88,8 +88,8 @@ To reuse shared content, follow these steps:
    {{</* docs/shared source="tempo" lookup="common-introduction.md" version="latest" */>}}
    ```
 
-   For more information about the `docs/shared` shortcode parameters, refer to [docs/shared shortcode]({{< relref "../../shortcodes#docsshared" >}}).
+   For more information about the `docs/shared` shortcode parameters, refer to [docs/shared shortcode](https://grafana.com/docs/writers-toolkit/write/shortcodes#docsshared).
 
 1. Verify the include.
 
-   To review the changes to the documentation, refer to the steps in [Review your changes]({{< relref "../../../contribute-documentation#review-your-changes" >}})
+   To review the changes to the documentation, refer to the steps in [Review your changes](https://grafana.com/docs/writers-toolkit/contribute#review-your-changes)
```
Comment (jdbaldry) [inline docs/sources/write/reuse-content/reuse-shared-content/index.md:95]:
```suggestion
   To review the changes to the documentation, refer to the steps in [Review your changes](https://grafana.com/docs/writers-toolkit/contribute/#review-your-changes)
```

**Review by jdbaldry — COMMENTED** [review]
Missing some trailing slashes

**Review by imatwawana — APPROVED** [review]
Tested all the links manually and they worked!

**Pull request merged**